promtimer/cbstats.py

```diff
@@ -448,6 +449,74 @@ def get_user_log(cluster, user, password):
             logging.error('error: can\'t access cluster: {}'.format(err))
             return []
 
+class BackupStatsFiles(Source):
+    """
+    Represents a source of stats data that is a Prometheus instance running against
+    the stats data from cbbackupmgr runs.
+    """
+    def __init__(self, prometheus_tsdb_path, short_name, prometheus_port):
+        super(BackupStatsFiles, self).__init__(prometheus_port)
+        self._short_name = short_name
+        self._prometheus_tsdb_path = prometheus_tsdb_path
+        self._config = None
+
+    def short_name(self):
+        return self._short_name
+
+    def host(self):
+        return '127.0.0.1'
+
+    def maybe_start(self, log_dir):
+        """
+        Starts the Prometheus instance that serves stats for this source.
+        """
+        log_path = path.join(log_dir, 'prom-{}.log'.format(self._short_name))
```

Comment:
if we can assume 3.6 (such a bold assumption, I know :trollface: ) you can use f-strings

 `f'prom-{self._short_name}.log'`

and similar below